packages/swapper/src/swappers/thorchain/ThorchainSwapper.ts

```diff
@@ -1,26 +1,58 @@
-import { AssetId } from '@shapeshiftoss/caip'
+import { adapters, AssetId } from '@shapeshiftoss/caip'
 import { Asset, SupportedChainIds } from '@shapeshiftoss/types'
+import axios from 'axios'
 
 import {
   ApprovalNeededOutput,
   GetMinMaxInput,
   MinMaxOutput,
+  SwapError,
+  SwapErrorTypes,
   Swapper,
   SwapperType,
   Trade,
   TradeQuote,
   TradeResult,
   TradeTxs
 } from '../../api'
+import { MidgardResponse } from './types'
+
+export type ThorchainSwapperDeps = {
+  midgardUrl: string
+}
+
 export class ThorchainSwapper implements Swapper {
+  public supportedAssetIds: AssetId[] = []
+  deps: ThorchainSwapperDeps
+
+  constructor(deps: ThorchainSwapperDeps) {
+    this.deps = deps
+  }
+
+  async initialize() {
+    try {
+      const { data: responseData } = await axios.get<MidgardResponse[]>(this.deps.midgardUrl)
+
+      const supportedAssetIds = responseData.reduce<AssetId[]>((acc, midgardPool) => {
+        const assetId = adapters.poolAssetIdToAssetId(midgardPool.asset)
+        if (!assetId) return acc
+        acc.push(assetId)
+        return acc
+      }, [])
```

Comment:
This won't work, given that `responseData` is an object of the midgard api response, it will not return the assetId. It's going to return the entire midgard response object.